### ISSUE TITLE
Make the reference configurable

### ISSUE THREAD

**Issue by leoisl** [mbhall88/tbpore#35]
This is a request from Floriane and Pedro from our meeting on 18/05/2022.

H37Rv is not always the reference they might want describe the variants with respect to, so the reference has to be configurable (e.g. through the parameters). Right now it is always forced to be H37Rv.

**Comment by mbhall88** [comment]
As long as they're aware that using different references will mean they won't be able to do SNP distances between isolates with different references

**Comment by leoisl** [comment]
Yeah, this is error prone. I am wondering if we just leave for the users to
manage it or if we try to control for this issue, like adding the sha256 of
the reference fasta to the header of the consensus, and when running tbpore
cluster we can check if all consensus were generated against the same ref

**Comment by mbhall88** [comment]
@iqbal-lab I think we need to stress this issue. I don't think we want to be detecting when they're providing consensus sequences based on different references...

**Comment by iqbal-lab** [comment]
It's called tbpore, I don't see why we need to support different references.  Will follow up

**Comment by leoisl** [comment]
It is still a TB reference, but a different one. Sometimes they want to compare with an African (if I remember correctly) reference strain, not with H37Rv

**Comment by mbhall88** [comment]
Yep. But the genome will be a different size and therefore SNP distances won't be applicable

**Comment by leoisl** [comment]
So, should we reconsider this or should we close as not planned?

**Comment by mbhall88** [comment]
We can make it configurable - it shouldn't be too difficult right? And just make sure we add a disclaimer in the help menu and docs that when using a different reference you can't compare SNP distances to other samples using a different reference.

**Comment by leoisl** [comment]
It is not hard to make it configurable. We could also easily control that `tbpore cluster` is called with consensus that were generated from a same ref - just add the ref sha-256 or similar hash to the first fasta header of the consensus. Of course this control can't be enforced if the consensus was not made by `tbpore` but then we could just have a `--ignore-ref-check` flag

**Comment by mbhall88** [comment]
Sounds like a good idea to me
